### PR TITLE
fix: stop reporting success for an empty table

### PR DESCRIPTION
**Pull request by ConnorMoss02**
## Summary

`append_block` with `type: "table"` and no cell content returned `{ appended: true }` and created a table with no cells. Nothing told the caller. Since `update_block` rejects `affine:table`, that table could not be filled afterwards either, so the success was a dead end.

This was the second item in #322. #323 fixed the schema gap that made `tableData` unreachable; this closes the silent case now that passing cell content is actually possible.

## Changes

- `src/tools/docs.ts`: under `strict`, which is the default, a `table` with neither `tableData` nor `tableCellDeltas` now raises, in the same style as the neighbouring `tableData` dimension errors. The message names the escape hatch, so a deliberately empty table is still one argument away.
- `src/tools/docs.ts`: non-strict callers keep the existing behaviour and get a `warnings` entry naming the block, using the `warnings: string[]` convention already used elsewhere in the file.
- `tests/test-input-contracts.mjs`: asserts the strict rejection reaches no AFFiNE request, alongside the existing table-cell contract tests.
- `CHANGELOG.md`: entry under Unreleased > Fixed.

## Validation

```bash
npm ci
npm run ci
```

- build, package dry-run and manifest checks passed
- 28 fast tests passed
- confirmed load-bearing: with the source change reverted, `test-input-contracts.mjs` fails on the new assertion

## Backward compatibility

This is the one deliberate behavior change: `append_block({ type: "table", rows, columns })` with no cell data previously succeeded and now raises by default.

I took that direction because an empty table cannot be filled through the tool surface afterwards, so the previous result was not usable. `strict: false` keeps the old behaviour exactly, now with a warning attached. Every other `append_block` call is unaffected, and `append_markdown` always supplies `tableData`, so the markdown path does not hit this.

Happy to flip it to warn-only in both modes if you would rather not change a default.

## Summary by CodeRabbit

- **New Features**
  - Empty tables can now be created successfully, even without initial cell content.
  - A warning is provided when a table is created without cells, including guidance for filling them with `tableData`, `tableCellDeltas`, or `update_table_cell`.
  - Empty table cells can be populated after creation using `update_table_cell`.

- **Documentation**
  - Added changelog details describing empty-table creation and the associated warnings.

### CODE REVIEW

**Comment by ConnorMoss02** [comment]
First one is right and it mattered: `appendBlockInternal` built the warning but the `doc.append_block` receipt never copied it, so the non-strict path still returned plain success. The warning was unreachable. Fixed in f21703d, copied the same way `ownedIds` and `missing` are.

Second one is already covered. `assert.equal(requestCount, 0, "invalid table cell input must not reach AFFiNE")` is the line directly after the new `assert.rejects`, and it runs after every rejection in that block, so a regression that sent the request before raising would fail there. Adding a second count assertion would restate it.

One thing worth flagging rather than leaving implied: I could not add a fast test for the propagation itself. Every test that exercises a successful `append_block` is classified `integration` in `tests/test-suites.json` and needs a live AFFiNE, so the fast suite reaches the strict rejection but not the non-strict success path. The receipt change is covered by typecheck and inspection only. Happy to add an integration test if you would like one there.

`npm run ci` green, 28 fast tests.

**Review by DAWNCR0W — CHANGES_REQUESTED** [review]
Thanks for the PR. I don’t think we should merge this as-is: `update_table_cell` can already populate an empty table, and the current schema explicitly supports creating one.

Please keep this warning-only, update the message and changelog to point users to `tableData`/`tableCellDeltas` or `update_table_cell`, and add a small integration test for creating an empty table and then filling a cell. The existing `requestCount` assertion already covers the unresolved CodeRabbit comment.

If there’s no update by Sep 8, I’ll take over the fix so we can move this forward.

**Comment by ConnorMoss02** [comment]
`update_table_cell` fills an empty table, so the rejection was wrong. Revised to warning-only.

- Strict throw removed; an empty table appends exactly as before.
- Warning reads: `Table <id> was created with no cell content. Pass tableData or tableCellDeltas on append_block to fill it, or use update_table_cell to set cells afterwards.`
- Changelog moved from Fixed to Added.
- `test-input-contracts.mjs` is back to its base state, so the existing `requestCount` assertion is untouched.
- `test-block-editing.mjs`: an empty table appends with the warning, a table given `tableData` carries none, then `update_table_cell` fills cell (1,1) from empty.

Ran against a live AFFiNE through the e2e stack.

**Comment by DAWNCR0W** [comment]
Thanks for the fix and the test coverage!

File: tests/test-input-contracts.mjs [hunk 1]
```diff
@@ -175,6 +175,11 @@ await assert.rejects(
   /The 'tableCellDeltas' field can only be used with type='table'/,
   "append_block must reject tableCellDeltas on a non-table block",
 );
+await assert.rejects(
+  appendBlock({ docId: "doc-1", type: "table", rows: 2, columns: 2 }),
+  /A table needs tableData \(or tableCellDeltas\) to have any cell content/,
+  "append_block must not silently create an empty table under strict",
+);
```
[the remark this answers is not in the document]
Comment (DAWNCR0W) [reply tests/test-input-contracts.mjs]: The strict-rejection test was removed in 9e44d97. This file now matches the base, including the existing requestCount assertion.
